spec/base_spec.rb

```diff
@@ -2,169 +2,170 @@
 
 require File.expand_path("../support", __FILE__)
 
-
 class BaseTest < CouchbaseOrm::Base
-    attribute :name, :string
-    attribute :job, :string
+  attribute :name, :string
+  attribute :job, :string
+  attribute(:prescribing_date, type: String, read_fn: proc { |value| encode_date(value) }) # timestamp without time zone,
+
+  class << self
+    def encode_date(value)
+      return DateTime.strptime(value, '%Y-%m-%d') if value.present? && value.is_a?(String) && value.length == 10
+      return DateTime.strptime(value, '%Y-%m-%d %H:%M:%s %z') if value.present? && value.is_a?(String)
+      value
+    end
+  end
 end
 
 class CompareTest < CouchbaseOrm::Base
-    attribute :age, :integer
+  attribute :age, :integer
 end
 
 class TimestampTest < CouchbaseOrm::Base
     attribute :created_at, :datetime
 end
 
+class TypeNamedTest < CouchbaseOrm::Base
+  attribute :count
+end
+
+
 describe CouchbaseOrm::Base do
-    it "should be comparable to other objects" do
-        base  = BaseTest.create!(name: 'joe')
-        base2 = BaseTest.create!(name: 'joe')
-        base3 = BaseTest.create!(ActiveSupport::HashWithIndifferentAccess.new(name: 'joe'))
-
-        expect(base).to eq(base)
-        expect(base).to be(base)
-        expect(base).not_to eq(base2)
-
-        same_base = BaseTest.find(base.id)
-        expect(base).to eq(same_base)
-        expect(base).not_to be(same_base)
-        expect(base2).not_to eq(same_base)
-
-        base.delete
-        base2.delete
-        base3.delete
-    end
+  it "should be comparable to other objects" do
```

Comment:
maybe it's me but seems' like most expect have been removed rending the test useless (but it's needed)